internal/api/external/user.go

```diff
@@ -186,7 +186,23 @@ func (a *UserAPI) Delete(ctx context.Context, req *pb.DeleteUserRequest) (*empty
 		return nil, grpc.Errorf(codes.Unauthenticated, "authentication failed: %s", err)
 	}
 
-	err := storage.DeleteUser(ctx, storage.DB(), req.Id)
+	sub, err := a.validator.GetSubject(ctx)
+	if err != nil {
+		return nil, helpers.ErrToRPCError(err)
+	}
+
+	if sub == auth.SubjectUser {
+		user, err := a.validator.GetUser(ctx)
+		if err != nil {
+			return nil, helpers.ErrToRPCError(err)
+		}
+
+		if (!user.IsAdmin || user.IsAdmin) && user.ID == req.Id {
```

Comment:
`(!user.IsAdmin || user.IsAdmin)`

Isn't that alway going to be true? :)

Reply:
Yes, I have added this for Admin as well as non-admin users but we can compare using `userid`
I will update this PR with test-case
Thanks

Reply:
But then there is no need to put `(!user.IsAdmin || user.IsAdmin)`?

E.g.

`user.IsAdmin = true` then it would mean `(false || true) = true`
`user.IsAdmin = false` then it would mean `(true || false) = true`

So `(!user.IsAdmin || user.IsAdmin)` is always `true`, so the only thing that really is validated is `user.ID == req.Id`

Reply:
I will remove `(!user.IsAdmin || user.IsAdmin)` and make new PR

Thanks for review

Reply:
No need to make a new PR, just push a new commit to this PR. I'll squash merge anyway, so additional commits are not going to pollute the history.

Reply:
Okay👍